test/algorithms/simulation/sidb/exhaustive_ground_state_simulation.cpp

```diff
@@ -21,57 +21,93 @@ TEMPLATE_TEST_CASE(
     {
         TestType lyt{{20, 10}};
 
+        lyt.assign_cell_type({0, 0, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({5, 0, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({7, 0, 0}, TestType::cell_type::NORMAL);
+
+        lyt.assign_cell_type({11, 0, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({13, 0, 0}, TestType::cell_type::NORMAL);
+
+        lyt.assign_cell_type({17, 0, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({19, 0, 0}, TestType::cell_type::NORMAL);
+
         exgs_stats<TestType>             exgs_stats{};
         const sidb_simulation_parameters params{2, -0.32};
         exhaustive_ground_state_simulation<TestType>(lyt, params, &exgs_stats);
+        auto size_before = exgs_stats.valid_lyts.size();
+        exhaustive_ground_state_simulation<TestType>(lyt, params, &exgs_stats);
+        auto size_after = exgs_stats.valid_lyts.size();
+        CHECK(size_before == 1);
+        CHECK(size_after == 1);
+
+        CHECK(exgs_stats.valid_lyts[0].get_charge_state({0, 0, 0}) == sidb_charge_state::NEGATIVE);
+        CHECK(exgs_stats.valid_lyts[0].get_charge_state({5, 0, 0}) == sidb_charge_state::NEUTRAL);
+        CHECK(exgs_stats.valid_lyts[0].get_charge_state({7, 0, 0}) == sidb_charge_state::NEGATIVE);
+        CHECK(exgs_stats.valid_lyts[0].get_charge_state({11, 0, 0}) == sidb_charge_state::NEUTRAL);
+        CHECK(exgs_stats.valid_lyts[0].get_charge_state({13, 0, 0}) == sidb_charge_state::NEGATIVE);
+        CHECK(exgs_stats.valid_lyts[0].get_charge_state({17, 0, 0}) == sidb_charge_state::NEUTRAL);
+        CHECK(exgs_stats.valid_lyts[0].get_charge_state({19, 0, 0}) == sidb_charge_state::NEGATIVE);
+        CHECK(std::abs(exgs_stats.valid_lyts[0].get_system_energy() - 0.24602741408) <
+              physical_constants::POP_STABILITY_ERR);
+    }
 
-        CHECK(exgs_stats.valid_lyts.size() == 0);  // empty layout is saved.
+    SECTION("simulation of SiQAD-OR gate with input 01")
+    {
+        TestType lyt{{20, 10}};
+
+        lyt.assign_cell_type({6, 2, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({8, 3, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({12, 3, 0}, TestType::cell_type::NORMAL);
+
+        lyt.assign_cell_type({14, 2, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({10, 5, 0}, TestType::cell_type::NORMAL);
+
+        lyt.assign_cell_type({10, 6, 1}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({10, 8, 1}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({16, 1, 0}, TestType::cell_type::NORMAL);
+
+        exgs_stats<TestType>             exgs_stats{};
+        const sidb_simulation_parameters params{2, -0.28};
+        exhaustive_ground_state_simulation<TestType>(lyt, params, &exgs_stats);
+
+        CHECK(exgs_stats.valid_lyts[0].get_charge_state({6, 2, 0}) == sidb_charge_state::NEGATIVE);
```

Comment:
For all tests, you should `REQUIRE` that the vector is not empty before accessing it.